mesonbuild/modules/i18n.py

```diff
@@ -360,11 +361,19 @@ def itstool_join(self, state: 'ModuleState', args: T.List['TYPE_var'], kwargs: '
         command: T.List[T.Union[str, build.BuildTarget, build.CustomTarget,
                                 build.CustomTargetIndex, 'ExternalProgram', mesonlib.File]] = []
         command.extend(state.environment.get_build_command())
+
+        # shlex.join() was added in Python 3.8
+        itstool_cmd = self.tools['itstool'].get_command()
+        if hasattr(shlex, 'join'):
+            itstool_cmd_str = shlex.join(itstool_cmd)
+        else:
+            itstool_cmd_str = ' '.join(shlex.quote(c) for c in itstool_cmd)
+
         command.extend([
             '--internal', 'itstool', 'join',
             '-i', '@INPUT@',
             '-o', '@OUTPUT@',
-            '--itstool=' + self.tools['itstool'].get_path(),
+            '--itstool="%s"' % itstool_cmd_str,
```

Comment:
This looks like it will fail. The point of shlex was to get a parseable string, why insert another pair of quotes?

Just change the value passed via the plus sign.